docs/frontend-performance.md

```diff
@@ -0,0 +1,2984 @@
+
+#### 6. Computations Dependent on Refs or DOM Values
+
+When computations depend on `useRef` values, DOM queries, or other mutable values, React Compiler cannot track changes statically.
+
+```typescript
+❌ WRONG: React Compiler can't optimize ref-based computations
+const TokenInput = ({ tokens }: Props) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [filter, setFilter] = useState('');
+
+  // React Compiler can't track ref.current changes statically
+  const filteredTokens = tokens.filter(token => {
+    const inputValue = inputRef.current?.value || filter;
+    return token.symbol.includes(inputValue);
+  });
+};
+
+✅ CORRECT: Manual memoization for ref-dependent computations
+const TokenInput = ({ tokens }: Props) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [filter, setFilter] = useState('');
+
+  // Manual memoization required - refs are mutable
+  const filteredTokens = useMemo(() => {
+    const inputValue = inputRef.current?.value || filter;
+    return tokens.filter(token => token.symbol.includes(inputValue));
+  }, [tokens, filter]); // Note: ref.current not in deps (intentional)
+};
+```
+
+**Why:** Refs are mutable values that React Compiler cannot track statically. DOM queries and other runtime values also cannot be analyzed at compile time.
+
+#### 7. Reselect Selectors and Complex Compositions
+
+When using Reselect selectors defined in other files, React Compiler cannot optimize the selector itself.
+
+```typescript
+❌ WRONG: React Compiler can't optimize Reselect selectors from other files
+// selectors.ts (different file)
+export const selectTotalBalance = createSelector(
+  [getAccounts, getBalances],
+  (accounts, balances) => accounts.reduce((sum, acc) =>
+    sum + balances[acc.address], 0
+  )
+);
+
+// component.tsx
+import { selectTotalBalance } from './selectors';
+
+const Dashboard = () => {
+  const accounts = useSelector(getAccounts);
+  const balances = useSelector(getBalances);
+
+  // React Compiler can't see into selectTotalBalance from another file
+  const totalBalance = accounts.reduce((sum, acc) =>
+    sum + balances[acc.address], 0
+  );
+};
+
+✅ CORRECT: Use Reselect selector (already memoized) or manual memoization
+// Option 1: Use Reselect selector (preferred - already memoized)
+const Dashboard = () => {
+  const totalBalance = useSelector(selectTotalBalance); // Reselect handles memoization
+};
+
+// Option 2: Manual memoization if selector not available
+const Dashboard = () => {
+  const accounts = useSelector(getAccounts);
+  const balances = useSelector(getBalances);
+
+  const totalBalance = useMemo(
+    () => accounts.reduce((sum, acc) => sum + balances[acc.address], 0),
+    [accounts, balances]
+  );
+};
+```
+
+**Why:** Reselect selectors defined in other files are opaque to React Compiler. However, Reselect already provides memoization, so this is often not an issue.
+
+#### 8. Effect Dependencies (Keep Existing Memoization)
```

Comment:
This is somehow repeating introduction from `React Compiler Limitations` section. I suggest we keep it at one place, more condensed, following DRY principle.